test/jdk/javax/swing/text/html/HtmlFontSizeTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/*
+ * @test
+ * @bug 8231286
+ * @key headful
+ * @summary  Verifies if HTML font size too large with high-DPI scaling and W3C_LENGTH_UNITS
+ * @run main/othervm -Dsun.java2d.uiScale=1.0 HtmlFontSizeTest
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Rectangle;
+
+import javax.swing.JFrame;
+import javax.swing.JEditorPane;
+import javax.swing.SwingUtilities;
+import javax.swing.text.Document;
+import javax.swing.text.html.HTMLEditorKit;
+import javax.swing.text.html.StyleSheet;
+
+public class HtmlFontSizeTest {
+    static Rectangle w3cFrameSize, stdFrameSize;
```

Comment:
These fields are accessed from two different threads. Shall they be declared as `volatile` then?
Declaring each field on its own line would follow the Java Code Conventions.